### DISCUSSION TITLE
Compressor - set parameters into performance.ini

### DISCUSSION THREAD

**Discussion by frank-que** [probonopd/MiniDexed#292] [Q&A] [answered]
Hello,
taking a look at the code I saw the compressor has fixed values, is there any specific reason for this?

Frank

**Comment by dcoredump** [comment, marked as answer]
Currently: yes. The compressor was needed because the Synth_Dexed engine got too fast into digital saturation - especially when playing several notes at a time.

The fixed values helped to avoid this.

But of corse they can be changed (if someone likes to implement this - it's not very difficult and a good starting point to get into synth programming 😉).

**Reply by frank-que** [reply]
I'll try to do, but honestly seems too complex for me... :(

**Reply by frank-que** [reply]
@dcoredump 
Would you be kind enough to show me the part of the code where the compressor function is called?
I haven't found it, but I'm probably wrong

**Reply by dcoredump** [reply]
Ah, sorry... I forget that the compressor code is inside Synth_Dexed - so you cannot find it in MiniDexed.

[This](https://codeberg.org/dcoredump/Synth_Dexed/src/commit/28ef0da952b5c3b20204f14a56724290235b154d/src/dexed.h#L179) is the header where you can find the public methods to set the values:

```
#ifdef USE_DEXED_COMPRESSOR
    void setCompressor(bool comp);
    bool getCompressor(void);
    void setCompressorPreGain_dB(float32_t pre_gain);
    void setCompressorAttack_sec(float32_t attack_sec);
    void setCompressorRelease_sec(float32_t release_sec);
    void setCompressorThresh_dBFS(float32_t thresh_dBFS);
    void setCompressionRatio(float32_t comp_ratio);
    float32_t getCompressorPreGain_dB(void);
    float32_t getCompressorAttack_sec(void);
    float32_t getCompressorRelease_sec(void);
    float32_t getCompressorThresh_dBFS(void);
    float32_t getCompressionRatio(void);
#endif
```
So you have to call the methods within the related object.

**Reply by frank-que** [reply]
Thanks Holger,
now I understand the reason... if I remove "effect_compressor.cpp" and "effect_compressor.h" from the Makefile I have no errors.

**Reply by dcoredump** [reply]
Yes... I added the files for using them as a master effect later. But until now I had no time to do this.
